src/main/java/org/terasology/gooeyDefence/EnemyManager.java

```diff
@@ -171,10 +173,38 @@ public void spawnEnemy(int entranceNumber) {
      * @param enemy The enemy to destroy
      */
     public void destroyEnemy(EntityRef enemy) {
+        dropMoney(enemy);
         enemies.remove(enemy);
         enemy.destroy();
     }
 
+    /**
+     * Drops the amount of money an enemy had.
+     * Drops money in 5 unit increments. If the enemy has no value, then nothing is dropped
+     *
+     * @param enemy The enemy to drop money for
+     */
+    private void dropMoney(EntityRef enemy) {
+        if (enemy.hasComponent(ValueComponent.class)) {
+            Vector3f location = enemy.getComponent(LocationComponent.class).getWorldPosition();
+            int value = enemy.getComponent(ValueComponent.class).getValue();
+
+            /* Drop the money in instances of 5 */
+            while (value >= 5) {
+                EntityRef money = entityManager.create("GooeyDefence:Money");
+                money.send(new DropItemEvent(location));
```

Comment:
I suggest to set the value explicit instead of relying on the default value in the prefab.